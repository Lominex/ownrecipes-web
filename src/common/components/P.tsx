import classNames from 'classnames';

import '../css/p.css';

export interface IPProps {
  variant?: 'body1' | 'body2';
  className?: string;
  children: React.ReactNode;
}

const P: React.FC<IPProps> = (props: IPProps) => (
  <p className={classNames(props.variant ?? 'body1', props.className)}>{props.children}</p>
);

export default P;
