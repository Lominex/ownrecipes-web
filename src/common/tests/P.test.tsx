import P from '../components/P';
import renderer from 'react-test-renderer';

test('P string', () => {
  const component = renderer.create(
    <P>Some string test</P>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('P jsx', () => {
  const component = renderer.create(
    <P>
      <div className='foo jsx'>
        <span>Some string test</span>
      </div>
    </P>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('P variant body1', () => {
  const component = renderer.create(
    <P variant='body1'>Some string test</P>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('P variant body2', () => {
  const component = renderer.create(
    <P variant='body2'>Some string test</P>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
