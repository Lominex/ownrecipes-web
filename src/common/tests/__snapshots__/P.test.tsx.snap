// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`P jsx 1`] = `
<p
  className="body1"
>
  <div
    className="foo jsx"
  >
    <span>
      Some string test
    </span>
  </div>
</p>
`;

exports[`P string 1`] = `
<p
  className="body1"
>
  Some string test
</p>
`;

exports[`P variant body1 1`] = `
<p
  className="body1"
>
  Some string test
</p>
`;

exports[`P variant body2 1`] = `
<p
  className="body2"
>
  Some string test
</p>
`;
